src/main/java/atdd/path/web/UserController.java

```diff
@@ -0,0 +1,107 @@
+package atdd.path.web;
+
+import atdd.path.application.dto.CreateUserRequestView;
+import atdd.path.application.dto.SignInUserRequestView;
+import atdd.path.application.dto.UserResponseView;
+import atdd.path.auth.JwtTokenDTO;
+import atdd.path.auth.JwtTokenProvider;
+import atdd.path.domain.User;
+import atdd.path.repository.UserRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private UserRepository userRepository;
+    private JwtTokenProvider jwtTokenProvider;
+
+    public UserController(UserRepository userRepository, JwtTokenProvider jwtTokenProvider) {
+        this.userRepository = userRepository;
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @PostMapping
+    public ResponseEntity createUser(@RequestBody CreateUserRequestView view) {
+        User persistUser = userRepository.save(view.toUser());
+        return ResponseEntity.created(URI.create("/users/" + persistUser.getId()))
+                             .body(UserResponseView.of(persistUser));
+    }
+
+    @GetMapping
+    public ResponseEntity showUsers() {
+        List<User> users = new ArrayList<User>();
+
+        userRepository.findAll().forEach(users::add);
+
+        return ResponseEntity.ok().body(UserResponseView.listOf(users));
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity showUser(@PathVariable Long id) {
+        Optional<User> optionalUser = userRepository.findById(id);
+
+        return optionalUser.map(this::getUserResponse)
+                           .orElseGet(this::notFoundResponse);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity deleteUser(@PathVariable Long id) {
+        userRepository.deleteById(id);
+
+        return noContentResponse();
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity signInUser(@RequestBody SignInUserRequestView view) {
+        Optional<User> savedUser = userRepository.findByEmail(view.getEmail());
+
+        return savedUser.map(user -> getTokenOrUnAuthResponse(user, view))
+                        .orElseGet(this::noContentResponse);
+    }
```

Comment:
```suggestion
    @PostMapping("/login")
    public ResponseEntity signInUser(@RequestBody SignInUserRequestView view) {
        User user = userRepository.findByEmail(view.getEmail()).orElseThrow(() ->  new EntityNotFoundException(view.getEmail() + "에 해당하는 유저가 존재하지 않습니다."));

        return ResponseEntity.ok(generateToken(user, view.getPassword()));
    }

    private JwtTokenDTO generateToken(User user, String password) {
        if (user.samePassword(password)) {
            return jwtTokenProvider.createToken(user.getEmail());
        }

        throw new IllegalArgumentException(user.getName() + "의 패스워드가 맞지 않습니다.");
    }
```

1. 실패시에 어떤 응답을 리턴하는지가 불분명합니다. (204인지, 403인지)
  예외처리를 하고 핸들링하는 것은 어떨까요

---

2. get method, 비즈니스 로직의 노출, ResponseEntity return 등은 다른 피드백에서 드렸으니 생략할게요~

---

3. 값을 하드코딩하기보다는 상수로 관리하는 것이 유지보수하기에 좋다고 생각합니다.
(예외처리의 문구도 물론 Exception을 관리하는 쪽에서 유지보수하는 것이 더 좋다고는 생각합니다.)
상수(static final)를 만들고 이름을 부여하면 이 변수의 역할이 무엇인지 의도를 드러내기에 좋다고 생각합니다.
자바 컨벤션 9.6 Constants를 확인해보세요.

https://myeonguni.tistory.com/1596

---

4. 인자로 파라미터를 넘길 때도 정보의 집중, 책임의 범위 등도 함께 고민해주시는 것이 좋다고 생각합니다.